### PR TITLE
Ignore class attribute when dynamically binding attributes

### PR DESCRIPTION
**Pull request by EWhite613**
Simple fix for https://github.com/DockYard/ember-one-way-controls/issues/80

### CODE REVIEW

File: addon/-private/dynamic-attribute-bindings.js [hunk 1]
```diff
@@ -8,11 +8,10 @@ export default Mixin.create({
 
     let newAttributeBindings = [];
     for (let key in this.attrs) {
-      if (this.NON_ATTRIBUTE_BOUND_PROPS.indexOf(key) === -1 && this.attributeBindings.indexOf(key) === -1) {
+      if (this.NON_ATTRIBUTE_BOUND_PROPS.indexOf(key) === -1 && this.attributeBindings.indexOf(key) === -1 && key !=='class') {
```
Comment (rwjblue) [inline addon/-private/dynamic-attribute-bindings.js]: I would think we should just put `class` in the `NON_ATTRIBUTE_BOUND_PROPS` array...

Comment (martndemus) [inline addon/-private/dynamic-attribute-bindings.js]: My thoughts exactly and then add `NON_ATTRIBUTE_BOUND_PROPS` to `concatenatedProperties`

**Comment by rwjblue** [comment]
I think we also need a test to ensure we don't regress in the future...

File: addon/-private/dynamic-attribute-bindings.js [hunk 2]
```diff
@@ -12,7 +12,6 @@ export default Mixin.create({
         newAttributeBindings.push(key);
       }
     }
-
     set(this, 'attributeBindings', this.attributeBindings.concat(newAttributeBindings));
```
Comment (martndemus) [inline addon/-private/dynamic-attribute-bindings.js]: Can you add to the mixin:
```js
NON_ATTRIBUTE_BOUND_PROPS: ['class'],
concatenatedProperties: ['NON_ATTRIBUTE_BOUND_PROPS']
```

File: addon/-private/dynamic-attribute-bindings.js [hunk 3]
```diff
@@ -12,7 +12,6 @@ export default Mixin.create({
         newAttributeBindings.push(key);
       }
     }
-
```
Comment (martndemus) [inline addon/-private/dynamic-attribute-bindings.js]: Please reinstate this empty line!

File: addon/components/one-way-checkbox.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ const OneWayCheckboxComponent = Component.extend(DynamicAttributeBindings, {
   tagName: 'input',
   type: 'checkbox',
 
-  NON_ATTRIBUTE_BOUND_PROPS: ['update'],
+  NON_ATTRIBUTE_BOUND_PROPS: ['update', 'class'],
```
Comment (martndemus) [inline addon/components/one-way-checkbox.js]: See `addon/-private/dynamic-attribute-bindings.js`

File: addon/components/one-way-input.js [hunk 5]
```diff
@@ -16,7 +16,8 @@ const OneWayInputComponent = Component.extend(DynamicAttributeBindings, {
   NON_ATTRIBUTE_BOUND_PROPS: [
     'keyEvents',
     'update',
-    'sanitizeInput'
+    'sanitizeInput',
+    'class'
   ],
```
Comment (martndemus) [inline addon/components/one-way-input.js]: See `addon/-private/dynamic-attribute-bindings.js` and remove `'class'` here.

File: addon/components/one-way-radio.js [hunk 6]
```diff
@@ -16,7 +16,8 @@ const OneWayRadioComponent = Component.extend(DynamicAttributeBindings, {
   NON_ATTRIBUTE_BOUND_PROPS: [
     'update',
     'option',
-    'value'
+    'value',
+    'class'
   ],
```
Comment (martndemus) [inline addon/components/one-way-radio.js]: See `addon/-private/dynamic-attribute-bindings.js` and remove `'class'` here.

File: addon/components/one-way-select.js [hunk 7]
```diff
@@ -30,7 +30,8 @@ const OneWaySelectComponent = Component.extend(DynamicAttributeBindings, {
     'includeBlank',
     'optionValuePath',
     'optionLabelPath',
-    'groupLabelPath'
+    'groupLabelPath',
+    'class'
```
Comment (martndemus) [inline addon/components/one-way-select.js]: See `addon/-private/dynamic-attribute-bindings.js` and remove `'class'` here.

File: tests/integration/components/one-way-checkbox-test.js [hunk 8]
```diff
@@ -49,3 +49,8 @@ test('It can accept an outside toggle of checked', function(assert) {
 
   assert.strictEqual(this.get('checked'), true);
 });
+
+test('I can add a class attribute', function(assert) {
+  this.render(hbs`{{one-way-checkbox class="testing"}}`);
+  assert.equal(true, this.$('input').attr('class').indexOf('testing') > -1);
```
Comment (martndemus) [inline tests/integration/components/one-way-checkbox-test.js]: jQuery has a `hassClass` function, so you could do `this.$('input').hasClass('testing')` instead.

**Comment by martndemus** [comment]
@EWhite613 thanks!
@homu r+

**Comment by martndemus** [comment]
Published v0.6.2 including this fix.

**Pull request merged**